src/sample/java/com/strangeberry/jmdns/tools/Browser.java

```diff
@@ -43,42 +48,40 @@
 import javax.swing.border.EmptyBorder;
 import javax.swing.event.ListSelectionEvent;
 import javax.swing.event.ListSelectionListener;
-import javax.swing.table.AbstractTableModel;
 
 /**
  * User Interface for browsing JmDNS services.
  *
  * @author Arthur van Hoff, Werner Randelshofer
  */
 public class Browser extends JFrame implements ServiceListener, ServiceTypeListener, ListSelectionListener {
-    /**
-     *
-     */
+
     private static final long serialVersionUID = 5750114542524415107L;
-    JmmDNS                    jmmdns;
-    // Vector headers;
-    String                    type;
-    DefaultListModel          types;
-    JList                     typeList;
-    DefaultListModel          services;
-    JList                     serviceList;
-    JTextArea                 info;
+    static JmmDNS jmmdns;
+    static MdnsBrowser instance;

```

Comment:
I think this should be `Browser`

```suggestion
    static Browser instance;
```